examples/sitemap/app/lib/sitemap.ts

```diff
@@ -0,0 +1,335 @@
+import {LoaderFunctionArgs} from '@shopify/remix-oxygen';
+import {
+  CountryCode,
+  LanguageCode,
+} from '@shopify/hydrogen/storefront-api-types';
+
+type Locale = `${LanguageCode}-${CountryCode}`;
+
+type SITEMAP_INDEX_TYPE =
+  | 'pages'
+  | 'products'
+  | 'collections'
+  | 'blogs'
+  | 'articles'
+  | 'metaObjects';
+
+/**
+ * Generate a sitemap index that links to separate sitemaps for each resource type.
+ */
+export async function getSitemapIndex({
+  storefront,
+  request,
+  types = [
+    'pages',
+    'products',
+    'collections',
+    'metaObjects',
+    'pages',
+    'articles',
+  ],
+  customUrls = [],
+}: {
+  storefront: LoaderFunctionArgs['context']['storefront'];
+  request: Request;
+  types?: SITEMAP_INDEX_TYPE[];
+  customUrls?: string[];
+}) {
+  const data = await storefront.query(SITEMAP_INDEX_QUERY, {
+    storefrontApiVersion: 'unstable',
+  });
+
+  if (!data) {
+    throw new Response('No data found', {status: 404});
+  }
+
+  const baseUrl = new URL(request.url).origin;
+
+  const body = `<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml">
+${types
+  .map((type) => getSiteMapLinks(type, data[type].pagesCount.count, baseUrl))
+  .join('\n')}
```

Comment:
It might be easier to read without template literals? e.g. `'...' + () + () + '...'` 🤔